graphql_api/schema/custom/rupture_generation_task.py

```diff
@@ -28,15 +30,20 @@
 log = logging.getLogger(__name__)
 
 
-class RuptureGenerationTask(graphene.ObjectType, AutomationTaskBase):
+class RuptureGenerationTask(AutomationTask):
     """An RuptureGenerationTask in the NSHM process"""
 
     class Meta:
         interfaces = (relay.Node, Thing, AutomationTaskInterface)
 
+    def resolve_task_type(root, info, **args):
+        if task_type := root.task_type:
+            return task_type
+        return TaskSubType.RUPTURE_SET
+
     @staticmethod
     def from_json(jsondata):
-        return RuptureGenerationTask(**AutomationTaskBase.from_json(jsondata))
+        return RuptureGenerationTask(**AutomationTask.from_json(jsondata))
```

Comment:
why does `RuptureGenerationTask` have a `from_json` method but `OpenquakeHazardTask` does not?